serverless/serverless
# add rollback feature

## What did you implement:

Closes #2412 

## How did you implement it:

I used the updateStack function of deploy. Therefor I moved it one level out. Before running updateStack there is function `setStackToUpdate`. This fetches the deployments from the S3 bucket. It verifies that the cloud formation template exists and only if proceeds with updateStack. In case the cloud formation template doesn't exist it will bail out early and show a nice error message.

## How can we verify it:

1. Create a service
2. Deploy the service
3. Invoke the function and verify the result.
4. Change the function output & deploy again
5. Invoke the function and verify the result.
6. List all deployments `serverless deploy list`
7. Take the first timestamp and run `serverless rollback -t <first timestamp>`
8. Invoke the function and verify the result.

## Todos:

- [x] Write integration tests

***Is this ready for review?:*** YES
